File: python/dec_01/dec_01.py
import sys
from typing import List

def first():
    with open("input_01.txt", "r") as input:
        lines = [line.strip() for line in input]

    result = 0
    dial = 50

    for line in lines:
        direction = line[0]
        number = int(line[1:])

        dial += number if direction == "R" else -number
        dial %= 100
        
        if dial == 0:
            result += 1

    print(f"1: {result}")

def second():
    with open("input_01.txt", "r") as input:
        lines = [line.strip() for line in input]

    result = 0
    dial = 50

    for line in lines:
        direction = line[0]
        number = int(line[1:])

        step = 1 if direction == "R" else -1
        
        for _ in range(number):
            dial = (dial + step) % 100
            if dial == 0:
                result += 1

    print(f"2: {result}")

first()
second()

File: ruby/dec_01/dec_01.rb
def first
  lines = File.readlines("input_01.txt").map(&:strip)

  result = 0
  dial = 50

  lines.each do |line|
    direction = line[0]
    number = line[1..].to_i

    dial += direction == "R" ? number : -number
    dial %= 100

    result += 1 if dial == 0
  end

  puts "first: #{result}"
end

def second
  lines = File.readlines("input_01.txt").map(&:strip)

  result = 0
  dial = 50

  lines.each do |line|
    direction = line[0]
    number = line[1..].to_i

    step = direction == "R" ? 1 : -1

    number.times do
      dial = (dial + step) % 100
      result += 1 if dial == 0
    end
  end

  puts "second: #{result}"
end

first
second